diff --git a/pkg/web/dojo_c_bg.js b/pkg/web/dojo_c_bg.js
index fd9bdec784f39abda4f3987b66467737910a4dbe..c866058eb2dcef64c0785a59af1d2a9ea1f3aa95 100644
--- a/pkg/web/dojo_c_bg.js
+++ b/pkg/web/dojo_c_bg.js
@@ -1,3 +1,24 @@
+
+
+/* thiscaspar: Patch to make torii-client work on nodejs */
+import crypto from 'crypto'
+import {performance} from 'perf_hooks'
+
+class Window {
+}
+
+const self = {
+    self:
+        Object.assign(new Window(), {
+            crypto: {
+                getRandomValues: buffer => crypto.randomFillSync(buffer)
+            },
+            performance
+        })
+};
+
+/* End of patch */
+
 let wasm;
 export function __wbg_set_wasm(val) {
     wasm = val;
